### PR TITLE
refactor documentation

### PR DESCRIPTION
**Pull request by aaguinal**
This PR updates the documentation found on https://algebraicjulia.github.io/AlgebraicRewriting.jl/dev/. 

More specifically, this:

- Provides a gentle overview of the package on the homepage
- Maintains the examples such that they are compatible with Catlab 0.16.5 and AlgebraicRewriting 0.3.0
- Compiles the library reference docs

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,12 +1,173 @@
...
+    IsMemberOf = [1, 2, 1, 2]
+
+    TeamName = ["Home", "Away"]
+    HasName = [1, 2]
+end
+l = ACSetTransformation(K, L, TeamName=[1, 2])
+r = ACSetTransformation(K, R, TeamName=[1, 2])
+```
+
+#### Colimit-of-representables instantiation (`@acset_colim``)
```
Comment (kris-brown) [inline docs/src/index.md:85]: extra backtick

File: docs/src/index.md [hunk 2]
```diff
@@ -1,12 +1,173 @@
...
+```
+
+### 7. Apply the rewrite rule 
+This executes the rewrite process using using the defined rule and match.
+
+```julia
+result = rewrite_match(rule, match)
+```
+
+This documentation provides a basic guide to using the AlgebraicRewriting package in Julia.
```
Comment (kris-brown) [inline docs/src/index.md:173]: Does this line belong at the beginning?

File: docs/src/index.md [hunk 3]
```diff
@@ -1,12 +1,173 @@
...
+- Right transformation, `r`, embed `K` in `R`.
+
+To define a rule, all five parts need to be defined. 
+
+It is possible to insert data according to the schema using a **static** approach _or_ the **colimit-of-representables** approach.
+
+#### Static Instantiation (`@acset`)
+If using the **static** approach, you must fully specify the ACSet functors and natural transformation. Here is a rule that defines the ACSet statically. 
+
+In this example, the rule swaps player between two teams.
```
Comment (kris-brown) [inline docs/src/index.md:59]: swaps player**s**?

Also the rule doesn't swap player #1 and #4, not sure if that's intended.

**Review by kris-brown — COMMENTED** [review]
Thanks so much Angeline, this is very helpful! I look forward to keeping up with the docs for new features in the future! I noticed the unicode art in the docstrings gets mangled, but happy to address that later.

**Comment by kris-brown** [comment]
also I suspect that `src/generated` should be added to the `.gitignore`

File: docs/src/index.md (hunk 3 above)
Comment (aaguinal) [reply docs/src/index.md:59]: Yeah, that is. It was intended to simulate a trade of players, one from each team. I revised the sentence to hopefully make it clearer.

File: docs/src/index.md (hunk 2 above)
Comment (aaguinal) [reply docs/src/index.md:173]: Yeah, you're right. It was a straggler. Moved it up.

File: docs/src/index.md (hunk 1 above)
Comment (aaguinal) [reply docs/src/index.md:85]: Fixed!

**Comment by kris-brown** [comment]
Looks great to me, thanks!

File: docs/make.jl [hunk 4]
```diff
@@ -1,25 +1,66 @@
...
-  target = "build",
-  repo   = "github.com/AlgebraicJulia/AlgebraicRewriting.jl.git",
-  branch = "gh-pages",
-  devbranch = "main"
-)
\ No newline at end of file
+  target="build",
+  repo="github.com/AlgebraicJulia/AlgebraicRewriting.jl.git",
+  branch="gh-pages",
+  devbranch="dev"
```
Comment (mehalter) [inline docs/make.jl:63]: This should not be `dev`, this should be `main`

**Pull request merged**